src/components/HomePage/PlayRadioButton.tsx

```diff
@@ -1,55 +1,79 @@
+/* eslint-disable react/no-multi-comp */
+import { useContext } from 'react';
+
+import { useActor, useSelector } from '@xstate/react';
 import useTranslation from 'next-translate/useTranslation';
-import { shallowEqual, useDispatch, useSelector } from 'react-redux';
 
 import PauseIcon from '../../../public/icons/pause.svg';
 import PlayIcon from '../../../public/icons/play-arrow.svg';
-import { triggerPauseAudio } from '../AudioPlayer/EventTriggers';
 import Button from '../dls/Button/Button';
 import Link from '../dls/Link/Link';
-import useCurrentStationInfo from '../Radio/useStationInfo';
+import Spinner from '../dls/Spinner/Spinner';
+import { getRandomCuratedStationId } from '../Radio/curatedStations';
+import { StationType } from '../Radio/types';
 
 import styles from './PlayRadioButton.module.scss';
 
-import { playFrom, selectIsPlaying, selectIsRadioMode } from 'src/redux/slices/AudioPlayer/state';
-import { selectRadioStation } from 'src/redux/slices/radio';
 import { logEvent } from 'src/utils/eventLogger';
+import { AudioPlayerMachineContext } from 'src/xstate/AudioPlayerMachineContext';
+import useCurrentStationInfo from 'src/xstate/Radio/useCurrentStationInfo';
+
+const RadioInformation = ({ radioActor }) => {
```

Comment:
This should be put in a separate file and `/* eslint-disable react/no-multi-comp */` line should be removed